tests/Integration/Adapter/ConfigurationTest.php

```diff
@@ -0,0 +1,230 @@
+
+        $this->assertEquals($expectedResult, $result);
+
+        if ($isLastTest) {
+            $this->cleanDb();
+        }
+    }
+
+    /**
+     * @return array[]
+     */
+    public function getProvider(): array
+    {
+        return [
+            // simple case: get an all shop config value
+            [
+                [
+                    [
+                        'key' => 'key_test_1',
+                        'value' => 'value_test_1',
+                        'shopConstraint' => ShopConstraint::allShops(),
+                    ],
+                ],
+                [
+                    'key' => 'key_test_1',
+                    'default' => false,
+                    'shopConstraint' => ShopConstraint::allShops(),
+                ],
+                'value_test_1',
+                false,
+            ],
+            // set and get a value for shop 2
+            [
+                [
+                    [
+                        'key' => 'key_test_1',
+                        'value' => 'value_test_1',
+                        'shopConstraint' => ShopConstraint::shop(2),
+                    ],
+                ],
+                [
+                    'key' => 'key_test_1',
+                    'default' => false,
+                    'shopConstraint' => ShopConstraint::shop(2),
+                ],
+                'value_test_1',
+                false,
+            ],
+            // try to get a non existing value for shop 2, get default value instead
+            [
+                [],
+                [
+                    'key' => 'does_not_exist',
+                    'default' => 'default_value',
+                    'shopConstraint' => ShopConstraint::shop(2),
+                ],
+                'default_value',
+                false,
+            ],
+            // try to get a non existing value for group 1, get default value instead
+            [
+                [],
+                [
+                    'key' => 'does_not_exist',
+                    'default' => 'default_value',
+                    'shopConstraint' => ShopConstraint::shopGroup(1),
+                ],
+                'default_value',
+                false,
+            ],
+            // get value for shop 2, inherited from parent group shop
+            [
+                [
+                    [
+                        'key' => 'parent_group_key_2',
+                        'value' => 'parent_group_value',
+                        'shopConstraint' => ShopConstraint::shopGroup(1),
+                    ],
+                ],
+                [
+                    'key' => 'parent_group_key_2',
+                    'default' => false,
+                    'shopConstraint' => ShopConstraint::shop(2),
+                ],
+                'parent_group_value',
+                false,
+            ],
+            // try getting a non existing value for shop 2, with is strict = true => should not inherit from parent group
```

Comment:
You also need to test getting a group value with strict
And you should set the value in the global context or the strict value has less interest